tests/test_unittest/test_gtc/test_gtir.py

```diff
@@ -91,43 +96,179 @@ def test_naive_python_copy(copy_computation):
-                                        )
-                                    )
-                                ],
+        vertical_loops=[
+            VerticalLoop(
+                loop_order=LoopOrder.FORWARD,
+                vertical_intervals=[
+                    VerticalInterval(
+                        start=AxisBound(level=LevelMarker.START, offset=0),
+                        end=AxisBound(level=LevelMarker.END, offset=0),
+                        body=[
+                            ParAssignStmt(
+                                left=FieldAccess.centered(name="a"),
+                                right=BinaryOp(
+                                    left=FieldAccess(
+                                        name="b",
+                                        offset=CartesianOffset(i=-1, j=0, k=0),
+                                    ),
+                                    right=FieldAccess(
+                                        name="b", offset=CartesianOffset(i=1, j=0, k=0)
+                                    ),
+                                    op=ArithmeticOperator.ADD,
+                                ),
                             )
                         ],
                     )
-                ]
+                ],
             )
         ],
     )
     assert ast.parse(PythonNaiveCodegen.apply(horizontal_avg))
+
+
+def test_ExprBaseclassIsNotInstantiatable():
+    with pytest.raises(TypeError):
+        Expr()
+
+
+def test_StmtBaseclassIsNotInstantiatable():
+    with pytest.raises(TypeError):
+        Stmt()
+
+
+def test_DeclBaseclassIsNotInstantiatable():
+    with pytest.raises(TypeError):
+        Decl()
+
+
+# Validation tests
+def test_ParAssignStmtWithVerticalOffsetIsOk():
+    ParAssignStmt(
+        left=FieldAccessBuilder("foo").offset(CartesianOffset(i=0, j=0, k=1))(), right=DummyExpr()
+    )
+
+
+def test_ParAssignStmtWithHorizontalOffsetIsError():
+    with pytest.raises(ValidationError):
+        ParAssignStmt(
+            left=FieldAccessBuilder("foo").offset(CartesianOffset(i=1, j=0, k=0))(),
+            right=DummyExpr(),
+        )
+
+
+arithmetic_type = DataType.FLOAT32
+another_arithmetic_type = DataType.INT32
+
+
+def test_TernaryOpValidNode():
+    assert (
+        TernaryOp(
+            cond=DummyExpr(dtype=DataType.BOOL),
+            true_expr=DummyExpr(dtype=arithmetic_type),
+            false_expr=DummyExpr(dtype=arithmetic_type),
+        ).dtype
+        == arithmetic_type
+    )
+
+
+def ternaryOpExprTypesMismatch():
+    return TernaryOp(
+        cond=DummyExpr(dtype=DataType.BOOL),
+        true_expr=DummyExpr(dtype=arithmetic_type),
+        false_expr=DummyExpr(dtype=another_arithmetic_type),
+    )
+
+
+def ternaryOpConditionIsNotBool():
+    return TernaryOp(
+        cond=DummyExpr(dtype=arithmetic_type),
+        true_expr=DummyExpr(),
+        false_expr=DummyExpr(),
+    )
+
+
+@pytest.fixture(params=[ternaryOpConditionIsNotBool, ternaryOpExprTypesMismatch])
```

Comment:
This would be the example with fixture. I think in this case it's the wrong pattern since we have to repeat the name (and the test `pytest.raises...` is trivial. If you have an idea how to improve the pattern, let me know.